strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout-provider/src/main/java/org/carlspring/strongbox/event/artifact/MavenArtifactDownloadedEventListener.java

```diff
@@ -0,0 +1,147 @@
+package org.carlspring.strongbox.event.artifact;
+
+import org.carlspring.maven.commons.util.ArtifactUtils;
+import org.carlspring.strongbox.client.ArtifactTransportException;
+import org.carlspring.strongbox.configuration.ConfigurationManager;
+import org.carlspring.strongbox.providers.ProviderImplementationException;
+import org.carlspring.strongbox.providers.layout.LayoutProvider;
+import org.carlspring.strongbox.providers.layout.LayoutProviderRegistry;
+import org.carlspring.strongbox.providers.repository.proxied.LocalStorageProxyRepositoryArtifactResolver;
+import org.carlspring.strongbox.providers.repository.proxied.ProxyRepositoryArtifactResolver;
+import org.carlspring.strongbox.providers.repository.proxied.SimpleProxyRepositoryArtifactResolver;
+import org.carlspring.strongbox.services.ArtifactMetadataService;
+import org.carlspring.strongbox.services.ArtifactResolutionService;
+import org.carlspring.strongbox.storage.repository.Repository;
+import org.carlspring.strongbox.storage.repository.RepositoryLayoutEnum;
+
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.NoSuchFileException;
+import java.security.NoSuchAlgorithmException;
+
+import org.apache.commons.io.FilenameUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.repository.metadata.Metadata;
+import org.apache.maven.index.locator.Locator;
+import org.apache.maven.index.locator.MetadataLocator;
```

Comment:
I'm not sure why you're using:
```
import org.apache.maven.artifact.repository.metadata.Metadata;
import org.apache.maven.index.locator.Locator;
```

These are used for rebuilding indexes, as far I'm aware.

Reply:
> import org.apache.maven.artifact.repository.metadata.Metadata;

is for rebuilding indexes ?

> import org.apache.maven.index.locator.Locator;
> import org.apache.maven.index.locator.MetadataLocator;

In these cases, you're right, I shouldn't have used them. I haven't noticed that. I'll use more proper tools for my requirements.

Reply:
> import org.apache.maven.artifact.repository.metadata.Metadata;
> 
> > is for rebuilding indexes?

No, this is fine. I was referring to the `import org.apache.maven.index.locator.*` imports.